src/layout/core/action.rs

```diff
@@ -0,0 +1,8 @@
+use rustwlc::{Point, ResizeEdge, WlcView};
+
+#[derive(Debug, Clone, Copy)]
+pub struct Action {
```

Comment:
This should be generic over some meta data `T`, so that the only thing associated with an action is the  `WlcView` and the `T`. In this case, we would make a new Action where `T`= `(Point, ResizeEdge)`